File: C/CS50/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);

int count_words(string text);

int count_sentences(string text);

//------------------------------------------------------------------------------------------------------------------------
int main(void)
{
    // Get text from user
    string text = get_string("Text: ");

    // Count the number of letters in the text
    int numLetters = count_letters(text);

    // Count the number of words in the text
    int numWords = count_words(text);

    // Count the number of sentences in the text
    int numSentences = count_sentences(text);

    // Calculate Coleman-Liau index
    double L = (double)numLetters / numWords * 100.0;
    double S = (double)numSentences / numWords * 100.0;

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    // Print the result
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}
//------------------------------------------------------------------------------------------------------------------------
int count_letters(string text)
{
    int lengthText = strlen(text);
    int letters = 0;

    for (int i = 0; i < lengthText; i++)
    {
        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            letters++;
        }
    }
    return letters;
}
//------------------------------------------------------------------------------------------------------------------------
int count_words(string text)
{
    int lengthText = strlen(text);
    int words = 1;

    for (int i = 0; i < lengthText; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}
//------------------------------------------------------------------------------------------------------------------------
int count_sentences(string text)
{
    int lengthText = strlen(text);
    int sentences = 0;

    for (int i = 0; i < lengthText; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Python/CS50/sentimental-readability/readability.py
# TODO

def main():
    # Get text from user
    text = input("Text: ")

    # Count the number of letters in the text
    numLetters = countLetters(text)

    # Count the number of words in the text
    numWords = countWords(text)

    # Count the number of sentences in the text
    numSentences = countSentences(text)

    # Calculate Coleman_Liau index
    l = float(numLetters / numWords * 100.0)
    s = float(numSentences / numWords * 100.0)

    index = round((0.0588 * l) - (0.296 * s) - 15.8)

    # Print result
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade", index)


def countLetters(text):
    letters = 0

    for i in range(len(text)):
        if (text[i] >= 'A' and text[i] <= 'Z') or (text[i] >= 'a' and text[i] <= 'z'):
            letters += 1

    return letters


def countWords(text):
    words = 1

    for i in range(len(text)):
        if text[i] == ' ':
            words += 1

    return words


def countSentences(text):
    sentences = 0

    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1

    return sentences


# Call main
main()